### PR TITLE
Add upload_file API for model and dataset

### PR DESCRIPTION
**Pull request by pulltheflower**
- Add upload_file API for model and dataset to support file upload

API path:

/api/v1/models/:namespace/:name/upload_file
/api/v1/datasets/:namespace/:name/upload_file

### CODE REVIEW

File: component/dataset.go [hunk 1]
```diff
@@ -559,3 +560,31 @@ func (c *DatasetComponent) UpdateDownloads(ctx context.Context, req *types.Updat
...
+	updateFileReq.NameSpace = req.NameSpace
+	updateFileReq.Name = req.Name
+	updateFileReq.FilePath = req.FilePath
+	updateFileReq.SHA = tree[0].SHA
+
+	_, err = c.UpdateFile(ctx, &updateFileReq)
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (Rader) [inline component/dataset.go:589]: return err

File: api/handler/model.go [hunk 2]
```diff
@@ -782,26 +780,24 @@ func (h *ModelHandler) UploadFile(ctx *gin.Context) {
...
-	fileBytes, err := io.ReadAll(openedFile)
-	if err != nil {
-		slog.Error("Error reading uploaded file", "error", err)
-		httpbase.BadRequest(ctx, err.Error())
-		return
-	}
-	encodedString := base64.StdEncoding.EncodeToString(fileBytes)
+	var buf bytes.Buffer
+	w := base64.NewEncoder(base64.StdEncoding, &buf)
+	io.Copy(w, openedFile)
```
Comment (Rader) [inline api/handler/model.go:785]: check error

File: api/handler/model.go [hunk 3]
```diff
@@ -782,26 +780,24 @@ func (h *ModelHandler) UploadFile(ctx *gin.Context) {
...
+
+	defer w.Close()
```
Comment (Rader) [inline api/handler/model.go:787]: should not defer

Comment (pulltheflower) [reply api/handler/model.go]: Done.

File: api/handler/model.go (hunk 2 above)
Comment (pulltheflower) [reply api/handler/model.go]: Done.

**Review by Rader — APPROVED** [review]
LGTM

**Pull request merged**